internal/factsengine/adapters/rabbitmq.go

```diff
@@ -62,18 +73,33 @@ func (r *RabbitMQAdapter) Listen(
 			}
 
 			return rabbitmq.Ack
-		},
-		queue,
-		[]string{routingKey},
-		rabbitmq.WithConsumeOptionsQueueDurable,
-		rabbitmq.WithConsumeOptionsBindingExchangeName(exchange),
-		rabbitmq.WithConsumeOptionsBindingExchangeKind("topic"),
-		rabbitmq.WithConsumeOptionsBindingExchangeDurable,
-	)
+		})
+		if err != nil {
+			log.Error(err)
+		}
+	}()
+
+	return nil
 }
 
-func (r *RabbitMQAdapter) Publish(exchange, routingKey, contentType string, message []byte) error {
-	return r.publisher.Publish(
+func (r *RabbitMQAdapter) Publish(
```

Comment:
Maybe we could pass the agent context here and use `PublishWithContext`.
With that, we could have `defer publisher.Close()` as well, and we wouldn't need to store the publisher anywhere, as we would always close after publishing.